### PR TITLE
feat: create MinimizableDialog component

### PR DESCRIPTION
**Pull request by shleewhite**
This PR includes the following changes:

- Creates a new package called @twilio-paste/callout that includes the following components:
  - MinimizableDialogContainer
  - MinimizableDialogButton
  - MinimizableDialog
  - MinimizableDialogHeader
  - MinimizableDialogContent
  - useMinimizableDialogState hook

### TODO
- [x] docs page
- [x] unit tests
- [x] customization unit tests

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: packages/paste-core/components/minimizable-dialog/src/MinimizableDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  'aria-label': string;
+}
+
+export const MinimizableDialog = React.forwardRef<HTMLDivElement, MinimizableDialogProps>(
+  ({children, element = 'MINIMIZABLE_DIALOG', ...props}, ref) => {
+    const dialog = React.useContext(MinimizableDialogContext);
+
+    return (
+      <NonModalDialogPrimitive
+        {...(dialog as any)}
```
Comment (SiTaggart) [inline packages/paste-core/components/minimizable-dialog/src/MinimizableDialog.tsx:39]: suggestion: I don't think you need this `as any` anymore. The way you extend a partial of non modal dialog primitive props for the context should do the trick. Try it out

Comment (shleewhite) [reply packages/paste-core/components/minimizable-dialog/src/MinimizableDialog.tsx:39]: Nice, that works!

File: packages/paste-core/components/minimizable-dialog/src/MinimizableDialogHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      i18nMinimizeLabel = 'minimize',
+      ...props
+    },
+    ref
+  ) => {
+    // @ts-ignore
+    // Property 'hide' does not exist on type 'Partial<PopoverState>'
+    // But reakit docs suggest using it
+    // https://reakit.io/docs/popover/#initial-focus
+    const {hide, minimized, minimize, expand} = React.useContext(MinimizableDialogContext);
```
Comment (SiTaggart) [inline packages/paste-core/components/minimizable-dialog/src/MinimizableDialogHeader.tsx:34]: praise: nice use of context and extending state to get these minimized states

File: packages/paste-core/components/minimizable-dialog/package.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "@twilio-paste/minimizable-dialog",
+  "version": "0.0.0",
+  "category": "overlays",
+  "status": "alpha",
```
Comment (SiTaggart) [inline packages/paste-core/components/minimizable-dialog/package.json:5]: chore: technically not alpha. Stage 1 yes, but this is prod spec

File: packages/paste-core/components/minimizable-dialog/src/MinimizableDialogButton.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  id?: string;
+  toggle?: () => void;
+};
+
+const MinimizableDialogButton = React.forwardRef<HTMLButtonElement, MinimizableDialogButtonProps>(
+  ({children, element = 'MINIMIZABLE_DIALOG_BUTTON', ...props}, ref) => {
+    const dialog = React.useContext(MinimizableDialogContext);
+
+    return (
+      <NonModalDialogDisclosurePrimitive element={element} {...(dialog as any)} {...props} as={Button} ref={ref}>
```
Comment (SiTaggart) [inline packages/paste-core/components/minimizable-dialog/src/MinimizableDialogButton.tsx:18]: question: what causes the need for `as any` here?

**Review by SiTaggart — APPROVED** [review]
Praise: honestly a couple of minor small things, but largely super shippable, so approving in advance of the last couple small things to get upated.

File: packages/paste-website/src/pages/components/minimizable-dialog/index.mdx [hunk 5]
```diff
@@ -0,0 +1,316 @@
+---
+title: Minimizable Dialog
+package: '@twilio-paste/minimizable-dialog'
+description: An accessible dialog that can be minimized to the bottom of the page.
```
Comment (gloriliale) [inline packages/paste-website/src/pages/components/minimizable-dialog/index.mdx:4]:
```suggestion
description: An accessible, non-modal dialog that can be minimized to the bottom of the page.
```

File: packages/paste-website/src/pages/components/minimizable-dialog/index.mdx [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+    Box,
+  }}
+  noInline
+>
+  {defaultExample}
+</LivePreview>
+
+## Guidelines
+
+### About Popover
```
Comment (gloriliale) [inline packages/paste-website/src/pages/components/minimizable-dialog/index.mdx:114]:
```suggestion
### About Minimizable Dialog
```

File: packages/paste-website/src/pages/components/minimizable-dialog/index.mdx [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+
+The Minimizable Dialog component is a dialog that can be minimized to the bottom right of the page. It is meant for situations like a chat dialog, where the user may start the chat and then want to minimize it while they complete other tasks on the page.
```
Comment (gloriliale) [inline packages/paste-website/src/pages/components/minimizable-dialog/index.mdx:116]:
```suggestion
The Minimizable Dialog component is a non-modal dialog that can be minimized to the bottom right of the page. It is meant for situations like a chat dialog, where the user may start the chat and then want to minimize it while they complete other tasks on the page.
```

File: packages/paste-website/src/pages/components/minimizable-dialog/index.mdx [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+to `MinimizableDialogContainer` as the `state` prop.
+
+This allows you to use certain returned props from the state hook, including functions like `hide` and `show`.
+
+It should be noted that when doing so, the state prop takes precedent over the other properties that affect
+the state or initial state of the Minimizable Dialog. They will be ignored in favour of them being provided as arguments
+to the `useMinimizableDialogState` hook.
+
+For full details on how to use the state hook, and what props to provide it, follow the
+[Non-Modal Dialog Primitive documentation](/primitives/non-modal-dialog-primitive).
```
Comment (gloriliale) [inline packages/paste-website/src/pages/components/minimizable-dialog/index.mdx:186]: praise: really like that we're linking them to the primitive docs here

**Review by gloriliale — APPROVED** [review]
This is STELLAR!

File: packages/paste-core/components/minimizable-dialog/package.json (hunk 3 above)
Comment (shleewhite) [reply packages/paste-core/components/minimizable-dialog/package.json:5]: oof then I should update that on callout too

File: packages/paste-core/components/minimizable-dialog/src/MinimizableDialogButton.tsx (hunk 4 above)
Comment (shleewhite) [reply packages/paste-core/components/minimizable-dialog/src/MinimizableDialogButton.tsx:18]: 👀 nothing, just that I copied it from popover. will fix.

File: packages/paste-core/components/minimizable-dialog/src/MinimizableDialogButton.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Button} from '@twilio-paste/button';
+import {NonModalDialogDisclosurePrimitive} from '@twilio-paste/non-modal-dialog-primitive';
+import {MinimizableDialogContext} from './MinimizableDialogContext';
+import type {ButtonProps} from '@twilio-paste/button';
+
+type MinimizableDialogButtonProps = Omit<ButtonProps, 'as'> & {
+  id?: string;
```
[the remark this answers is not in the document]
Comment (shleewhite) [reply packages/paste-core/components/minimizable-dialog/src/MinimizableDialogButton.tsx:9]: Yup! it is spread with the rest of the props.

**Pull request merged**